guide/07-working-with-bigdata/10-extending_geoanalytics_with_custom_tools.ipynb

```diff
@@ -0,0 +1,434 @@
+{
```

Comment:
<p>In the example below, the <a href="https://spark.apache.org/docs/2.2.0/ml-clustering.html#k-means" target="_blank">k-means</a> algorithm is used to segment a map of crime data into a predefined number of clusters. You can then use this map to help efficiently dispatch officers throughout a city. The k-means algorithm is not included in GeoAnalytics Tools; however, it is included in the pyspark.ml package. By calling this implementation of k-means in the Run Python Script tool, you can execute k-means on your GeoAnalytics Server site using an ArcGIS Enterprise layer as input. You can then save the result as a feature service or other ArcGIS Enterprise layer type.</p><p> </p><p><strong>Using k-means to cluster crime events with the pyspark.ml package.</strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1076/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>